### PR TITLE
Use generic default names instead of opinionated ones.

### PR DESCRIPTION
**Pull request by dhaus67**
## Description

When installing the observability operator, by default a `Observability` is created. This will include default settings.

Currently, the default settings will include the following, opinionated names:
- AlertManager will be named `kafka-alertmanager`.
- Grafana will be named `kafka-grafana`.
- Prometheus will be named `kafka-prometheus`.
- As label selectors, `app: strimzi` will be required.

It's especially unfortunate that these names are leaking also into customer configuration repositories (i.e. the label selectors).
Since the observability operator should be a generic one, we should strive to use generic names for these resources as label selectors.
For this, mostly all (will be tracking this in the follow-ups further below) resource and label selector names have been renamed from `kafka-xxx` to `observability-xxx`. This will work just fine for newly created `Observabilities`, but will impose a required migration for existing `Observabilities`, which created resources using the old default names.

To overcome this, the following has been introduced within this PR:
- Renaming default names.
- During initialization, check whether the resource (grafana, prometheus, alertmanager) is found with the existing configuration (taking the new default name into account). If it is not found _and_ is found with the old name, use the CR settings to override default names and set the values to the old default names.
- Within the validation webhook, allow updating the CR, iff the old default name is used.

What will be done in a follow-up PR:
- The promtail names are still opinionated. If we can agree on the migration, they would need a `defaultPromtailName` setting within the CR, this should be done in a separate PR as to not increase the scope of this one.

## Testing

Testing has been done manually as of now, I'm happy to add more automated tests either in this PR or a follow-up.
1. Create a new CR with updated default names - ensure the new default names are used and the application is working as intended.
2. Create a CR with old default names - after switching to the new operator version, ensure that reconciliation is successful and the CR is updated accordingly.
3. Create a CR with old default names - switch to a new operator version using updated default names - ensure deletion is successful and no dangling resources can be found.

### CODE REVIEW

**Comment by dhaus67** [comment]
@pb82 tagging you for visibility.

**Comment by vmanley** [comment]
@dhaus67 I was able to try out your change but unfortunately we are denied by webhook when the operator tries migrate the old default names
`2022-06-21T14:14:00.360Z	ERROR	controllers.Observability	failed to update operand to handle new default names	{"error": "admission webhook \"vobservability.kb.io\" denied the request: cannot set GrafanaDefaultName after cr creation"}`
There is a self-contained option for the operator (some detail found [here](https://github.com/redhat-developer/observability-operator#whats-supported-via-external-config) in README). Could this be useful for your use case?

File: api/v1/observability_webhook.go [hunk 1]
```diff
@@ -69,7 +69,9 @@ func (in *Observability) ValidateUpdate(old runtime.Object) error {
 	}
 
 	if oldObsSpec.AlertManagerDefaultName == "" &&
-		newObsSpec.AlertManagerDefaultName != "" {
+		newObsSpec.AlertManagerDefaultName != "" &&
+		// Special case: Allow updates iff the deprecated default name is used.
```
[the remark this answers is not in the document]
Comment (obrienrobert) [reply api/v1/observability_webhook.go:73]: Think the typo exists in a couple of other places as well.

**Comment by dhaus67** [comment]
@vmanley while this works for the label selector and such, what we previously discussed with Peter was that it'd be nice if the generic observability operator wouldn't have any opinionated names (i.e. any kafak / strimzi specific values). 

Currently, within our environment, we cannot take the `observability-operator-no-init` config map way of providing our custom CR instead of the default created one, thus ending up using the kafka-opinionated names for the resources. While the config option you mentioned will solve the label one, other resources such as the deployments will still inherit the kafka-opinionated names.

I've fixed the issued with the webhook error, as I've mistakenly used the `AlertManagerDefaultName` instead of the `GrafanaDefaultName` in the condition, sorry for the noise there.

**Comment by vmanley** [comment]
@dhaus67 thanks for the added detail and small correction. I'll have another look. We may need other significant changes to move to more generic default resource names but it would certainly be a worthwhile goal. Thanks!

**Comment by pb82** [comment]
@dhaus67 We've verified your changes but there have been a few issues:

1. The new resources are created, but the old ones don't get deleted
2. During cleanup, only the old resources are removed

I'm wondering if we can provide a simple strategy instead:

1. For new installations, always use the new neutral default names
2. For existing installations, do the same. To deal with the now duplicate resources, we implement a cleanup phase that just deletes the old ones.
3. To not run the cleanup phase every time, we add an annotation to the Observability CR once it's finished and skip it in future reconciliations.

What do you think about that? Anything i'm missing? The default name for the PVC is `managed-services`. Not ideal, do you think that should have a different name as well?

**Comment by pb82** [comment]
@dhaus67 we're exploring that strategy in #136

**Comment by dhaus67** [comment]
@pb82 
>For existing installations, do the same. To deal with the now duplicate resources, we implement a cleanup phase that just deletes the old ones.

Not sure I understand it correctly, but wouldn't this break existing configuration of clients that [use the label within their external config](https://github.com/redhat-developer/observability-operator#whats-supported-via-external-config)?

Other than that, this sounds sound to me.

>The default name for the PVC is managed-services.

I think its worthwhile to name this differently as well.

I saw the PR #136 , I _think_ I got the idea of it. How shall we proceed, should we just converge on that PR and step-by-step add the cleanup phase for all resource? I can help with that / take that over.

**Comment by pb82** [comment]
fixed with #136

**Pull request closed without merging**